lib/ansible/modules/cloud/vmware/vmware_host_config_facts.py

```diff
@@ -0,0 +1,109 @@
+---
+module: vmware_host_config_facts
+short_description: Gathers facts about an ESXi host's advance configuration information
+description:
+- This module can be used to gather facts about an ESXi host's advance configuration information when ESXi hostname or Cluster name is given.
+version_added: '2.5'
+author:
+- Abhijeet Kasurde (@akasurde)
+notes:
+- Tested on vSphere 6.5
+requirements:
+- python >= 2.6
+- PyVmomi
+options:
+  cluster_name:
+    description:
+    - Name of the cluster from which the ESXi host belong to.
+    - If C(esxi_hostname) is not given, this parameter is required.
+  esxi_hostname:
+    description:
+    - ESXi hostname to gather facts from.
+    - If C(cluster_name) is not given, this parameter is required.
+extends_documentation_fragment: vmware.documentation
+'''
+
+EXAMPLES = r'''
+- name: Gather facts about all ESXi Host in given Cluster
+  vmware_host_config_facts:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    cluster_name: cluster_name
+
+- name: Gather facts about ESXi Host
+  vmware_host_config_facts:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    esxi_hostname: '{{ esxi_hostname }}'
+'''
+
+RETURN = r'''#
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.vmware import vmware_argument_spec, PyVmomi
+
+
+class VmwareConfigFactsManager(PyVmomi):
+    def __init__(self, module):
+        super(VmwareConfigFactsManager, self).__init__(module)
+        self.cluster = self.params.get('cluster_name', None)
+        self.host = self.params.get('esxi_hostname', None)
+        self.hosts = []
+        if self.cluster:
+            self.cluster_obj = self.find_cluster_by_name(cluster_name=self.cluster)
+            if self.cluster_obj:
+                self.hosts = [host for host in self.cluster_obj.host]
+        elif self.host:
+            self.host_obj = self.find_hostsystem_by_name(host_name=self.host)
+            if self.host_obj:
+                self.hosts.append(self.host_obj)
+
+    def gather_host_facts(self):
+        hosts_facts = {}
+        for host in self.hosts:
+            host_facts = {}
+            for option in host.configManager.advancedOption.QueryOptions():
+                host_facts[option.key] = option.value
+            hosts_facts[host.name] = host_facts
+        return hosts_facts
+
+
+def main():
+    argument_spec = vmware_argument_spec()
+    argument_spec.update(
+        cluster_name=dict(type='str', required=False),
+        esxi_hostname=dict(type='str', required=False),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        required_one_of=[
+            ['cluster_name', 'esxi_hostname'],
+        ]
+    )
+
+    vmware_host_config = VmwareConfigFactsManager(module)
+    module.exit_json(changed=True, hosts_facts=vmware_host_config.gather_host_facts())
```

Comment:
`changed=False` (this module doesn't do any change).

Reply:
OK.